ci/utils.go

```diff
@@ -3,174 +3,255 @@ package ci
 import (
 	"bytes"
 	"context"
+	"crypto/rand"
+	"encoding/hex"
 	"encoding/json"
 	"flag"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"net/url"
+	"os"
+	"strings"
 	"testing"
 	"time"
 
+	xc "github.com/cordialsys/crosschain"
+	client "github.com/cordialsys/crosschain/client"
 	"github.com/cordialsys/crosschain/cmd/xc/handler"
 	"github.com/cordialsys/crosschain/cmd/xc/setup"
-	"github.com/spf13/cobra"
+	"github.com/cordialsys/crosschain/factory"
 	"github.com/test-go/testify/require"
 )
 
-type Tx struct {
-	Name          string     `json:"name"`
-	Hash          string     `json:"hash"`
-	Chain         string     `json:"chain"`
-	Block         BlockInfo  `json:"block"`
-	Movements     []Movement `json:"movements"`
-	Fees          []Fee      `json:"fees"`
-	Confirmations int        `json:"confirmations"`
+var (
+	chain string
+	rpc   string
+)
+
+type contextKey string
+
+const (
+	xcFactoryKey   contextKey = "xcFactory"
+	chainConfigKey contextKey = "chainConfig"
+	rpcKey         contextKey = "rpc"
+)
+
+func init() {
+	flag.StringVar(&chain, "chain", "", "Blockchain chain to use")
+	flag.StringVar(&rpc, "rpc", "", "RPC endpoint")
 }
 
-type BlockInfo struct {
-	Chain  string `json:"chain"`
-	Height int    `json:"height"`
-	Hash   string `json:"hash"`
-	Time   string `json:"time"`
+func validateRequiredFlags(t *testing.T, value, errorMsg string) {
+	if value == "" {
+		t.Fatal(errorMsg)
+	}
 }
 
-type Movement struct {
-	Asset    string    `json:"asset"`
-	Contract string    `json:"contract"`
-	AssetID  string    `json:"asset_id"`
-	From     []Address `json:"from"`
-	To       []Address `json:"to"`
+func initializeContext(chain, rpc string) (context.Context, error) {
+	rpcArgs := &setup.RpcArgs{
+		Chain:     chain,
+		Rpc:       rpc,
+		Overrides: map[string]*setup.ChainOverride{},
+	}
+
+	xcFactory, err := setup.LoadFactory(rpcArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	chainConfig, err := setup.LoadChain(xcFactory, rpcArgs.Chain)
+	if err != nil {
+		return nil, err
+	}
+
+	setup.OverrideChainSettings(chainConfig, rpcArgs)
+
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, xcFactoryKey, xcFactory)
+	ctx = context.WithValue(ctx, chainConfigKey, chainConfig)
+	ctx = context.WithValue(ctx, rpcKey, rpc)
+
+	return ctx, nil
```

Comment:
This is better, but I think we don't need to use this context.  We can just use concrete types.

Reply:
I've used this approach as this code is used in each function and I thought about placing it in a single method will be easy to maintain.